rllib/utils/spaces/space_utils.py

```diff
@@ -6,6 +6,31 @@
 from typing import Any, List, Optional, Union
 
 
+@DeveloperAPI
+class BatchedNdArray(np.ndarray):
+    """A simple ndarray-wrapped adding the _has_batch_dim boolean flag.
+
+    This is such that our `batch` utility can distinguish between having to
+    stack n individual batch items (each one w/o any batch dim) vs having to
+    concatenate n already batched items (each one possibly with a different batch
+    dim, but definitely with some batch dim).
+    """
+
+    def __new__(cls, input_array):
+        # Use __new__ to create a new instance of our subclass.
+        obj = np.asarray(input_array).view(cls)
+        # Set the _has_batch_dim property.
+        obj._has_batch_dim = True
```

Comment:
Why do we need this attribute? What I see is that we only check against an object being an instance of type `BatchedNdArray` and not if `_has_batch_dim`.

Reply:
I think you are right. This being another class should be sufficient.

Another option would be to signal via the `list`in which we are storing these that this is a list-to-be-concatenated (rather than a list-to-be-stacked). 🤔 

This would remove the messiness of the `BatchedNdArray` approach, in which we normally should check, whether really all items in the list are of that class (which we don't do right now!).

Reply:
Removed.